source/server/listener_manager_impl.cc

```diff
@@ -403,9 +403,20 @@ bool ListenerManagerImpl::addOrUpdateListenerInternal(
     return false;
   }
 
-  ListenerImplPtr new_listener(new ListenerImpl(config, version_info, *this, name, added_via_api,
-                                                workers_started_, hash,
-                                                server_.options().concurrency()));
+  ListenerImplPtr new_listener = nullptr;
+
+  // Intelligent listener update assumes there is active listener.
```

Comment:
per other comments "intelligent" is not descriptive. If we need language in the code/comments let's be specific and having naming around "in place filter chain update", "drainless update" or similar